src/Libraries/Microsoft.AspNetCore.Diagnostics.Middleware/Buffering/IHttpRequestBufferManager.cs

```diff
@@ -0,0 +1,20 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+#if NET9_0_OR_GREATER
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.Shared.DiagnosticIds;
+
+namespace Microsoft.Extensions.Diagnostics.Buffering;
+
+/// <summary>
+/// Interface for a HTTP request buffer manager.
+/// </summary>
+[Experimental(diagnosticId: DiagnosticIds.Experiments.Telemetry, UrlFormat = DiagnosticIds.UrlFormat)]
+public interface IHttpRequestBufferManager : IBufferManager
```

Comment:
Could this type be internal?

Reply:
Users will use buffering either via this interface (for HTTP or non-HTTP request logs or for both) or via `IBufferManager`(for non-HTTP request logs only) injected via constructor, that's why it should be `public`.